specs/ScrollingControls/ScrollPresenter.md

```diff
@@ -0,0 +1,2462 @@
+        Double horizontalOffset,
+        Double verticalOffset,
+        Microsoft.UI.Xaml.Controls.ScrollingScrollOptions options);
+    [method_name("ScrollBy")]
+    Int32 ScrollBy(
+        Double horizontalOffsetDelta,
+        Double verticalOffsetDelta);
+    [method_name("ScrollByWithOptions")]
+    Int32 ScrollBy(
+        Double horizontalOffsetDelta,
+        Double verticalOffsetDelta,
+        Microsoft.UI.Xaml.Controls.ScrollingScrollOptions options);
+    Int32 AddScrollVelocity(
+        Windows.Foundation.Numerics.Vector2 offsetsVelocity,
+        Windows.Foundation.IReference<Windows.Foundation.Numerics.Vector2> inertiaDecayRate);
+    [method_name("ZoomTo")]
+    Int32 ZoomTo(
+        Single zoomFactor,
+        Windows.Foundation.IReference<Windows.Foundation.Numerics.Vector2> centerPoint);
+    [method_name("ZoomToWithOptions")]
+    Int32 ZoomTo(
+        Single zoomFactor,
+        Windows.Foundation.IReference<Windows.Foundation.Numerics.Vector2> centerPoint,
+        Microsoft.UI.Xaml.Controls.ScrollingZoomOptions options);
+    [method_name("ZoomBy")]
+    Int32 ZoomBy(
+        Single zoomFactorDelta,
+        Windows.Foundation.IReference<Windows.Foundation.Numerics.Vector2> centerPoint);
+    [method_name("ZoomByWithOptions")]
+    Int32 ZoomBy(
+        Single zoomFactorDelta,
+        Windows.Foundation.IReference<Windows.Foundation.Numerics.Vector2> centerPoint,
+        Microsoft.UI.Xaml.Controls.ScrollingZoomOptions options);
+    Int32 AddZoomVelocity(
+        Single zoomFactorVelocity,
+        Windows.Foundation.IReference<Windows.Foundation.Numerics.Vector2> centerPoint,
+        Windows.Foundation.IReference<Single> inertiaDecayRate);
+
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Object> ExtentChanged;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Object> StateChanged;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Object> ViewChanged;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Microsoft.UI.Xaml.Controls.ScrollingScrollAnimationStartingEventArgs> ScrollAnimationStarting;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Microsoft.UI.Xaml.Controls.ScrollingZoomAnimationStartingEventArgs> ZoomAnimationStarting;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Microsoft.UI.Xaml.Controls.ScrollingScrollCompletedEventArgs> ScrollCompleted;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Microsoft.UI.Xaml.Controls.ScrollingZoomCompletedEventArgs> ZoomCompleted;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Microsoft.UI.Xaml.Controls.ScrollingBringingIntoViewEventArgs> BringingIntoView;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, Microsoft.UI.Xaml.Controls.ScrollingAnchorRequestedEventArgs> AnchorRequested;
+    event Windows.Foundation.TypedEventHandler<ScrollPresenter, MU_XCP_NAMESPACE.ScrollingViewChangeRequestedEventArgs> ViewChangeRequested;
```

Comment:
Is ScrollView going to listen to all of these events anyway? Would it be easier to make it a callback interface? We usually prefer events to that, but if the use case is to listen to everything, then it's more efficient.